ERCS/erc-7683.md

```diff
@@ -72,6 +72,47 @@ Cross-chain execution systems implementing this standard SHOULD use a sub-type t
 
 All sub-types SHOULD be registered at github.com/erc-7683/order-subtypes to encourage sharing of sub-types based on their functionality.
 
+### Decoding cross-chain orders
+
+A compliant cross-chain order type MUST be convertible into the `ResolvedCrossChainOrder` struct. This means that the `orderData` must be decoded into parameters that are used to populate the `ResolvedCrossChainOrder` struct. The decoded `orderData` SHOULD also contain data to construct a sub-type, which can be used for further functionality such as cross-chain calldata execution. For example, there might be a sub-type called `Message`:
+
+```
+Message
+
+The Message subtype allows ERC7683 intents to carry messages that are executed on the destination chain.
+
+Struct MessageInfo {
+  bytes message;
+  address fallbackTarget;
+}
+
+Sub-type selector: "Message"
+
+The intent recipient will receive the tokens, then the message will be called. If the fallbackTarget is set, then, in the case of revert, the tokens will be sent to the fallbackTarget. If not set, reverts will cause the entire transaction to fail and the intent will be impossible to fulfill.
+
+To use this extension, abi.encode the Message function (pre-pending the selector). Then include those bytes in the list of sub-types in the ERC 7683 orderData.
+```
+
+Then, the `orderData` could be decoded as follows within a `settlementContract`'s fill function:
+
+```solidity
+
+// Let's assume we have the original 7683 order's orderData:
+bytes memory orderData;
+
+(ResolvedCrossChainOrder memory resolvedCrossChainOrder, bytes[] subtypes) = abi.decode(orderData)
```

Comment:
I'm curious what you think @mrice32 about the best way to delimit between the parameters needed to fill out a ResolvedCrossChainOrder and anything for the subtype. We'd need to (1) know the type of subtype struct to look for and also where in the `bytes orderdata` to find this information.